Sources/SwiftCliCore/ChartConstructor.swift

```diff
@@ -47,40 +23,50 @@ public class ChartConstructor {
             let middleBoxes = String(repeating: "─┬", count: width - 1)
             return "┌\(middleBoxes)─┐\n"
         }
-
-    }
-    public func makeMiddleRow(width: Int) -> String {
-        switch width {
-        case 0:
-            return "\n"
-        case 1:
-            return "├─┤\n"
-        default:
-            let middleBoxes = String(repeating: "─┼", count: width - 1)
-            return "├\(middleBoxes)─┤\n"
-        }
     }
 
-    public func makeMiddleRowStitchCountChange(width: Int, rDiff: Int, lDiff: Int) -> String {
-        let leftStitches = "├"
+    public func makeMiddleRow(width: Int, rDiff: Int, lDiff: Int) -> String {
+        var leftStitches = ""
         var rightStitches = ""
         var middleStitches = ""
 
-        switch width {
-        case 1:
-            middleStitches = "─"
+        switch true {
+        case (lDiff == 1):
+            leftStitches = "└─┼"
+        case (lDiff >  1):
+            let midLeftSt = String(repeating: "─┴", count: ((abs(lDiff))-1))
+            leftStitches = "└\(midLeftSt)─┼"
         default:
-            middleStitches = String(repeating: "─┼", count: (width-1))
+            leftStitches = "├"
         }
 
+        var totalIncreases = 0
+
+        switch true {
+        case (lDiff > 0 && rDiff > 0):
+            totalIncreases += rDiff
+            totalIncreases += lDiff
+        case (rDiff > 0):
+            totalIncreases += rDiff
+        case (lDiff > 0):
+            totalIncreases += lDiff
+        default:
+            totalIncreases = 0
+        }
+
+        middleStitches = String(repeating: "─┼", count: (width-totalIncreases-1))
```

Comment:
Since you don't alter `middleStitches` any further, you don't need to initialize it as an empty string first.